File: protected/views/client/anamnese.php
<?php $form=$this->beginWidget('CActiveForm', array(
	'id'=>'anamnese-form',
	'enableAjaxValidation'=>false,
)); ?>
<style>
	.anamnese-table {width: 100%; border-collapse: collapse; }
	.anamnese-table td {width: 50%; border: 1px solid #000;padding: 10px;  }
	.anamnese-table th {background: #000; color: #fff; text-align: center; border: 1px solid #000; padding: 10px; }
	.anamnese-table textarea,  .anamnese-table input[type=text] {width: 100%; border-width: 0 0 1px 0; border-color: #000; }
	.anamnese-table ul {list-style-type: none; margin: 0;}
	.anamnese-table ul.labels li {height: 23px;}
	.anamnese-table ul.inputs li {height: 23px;}
	.anamnese-table .separator {display: inline-block; width: 33%;}
</style>
<h3 style="text-align: center;width:100%;">Anamnesefragebogen</h3>
<table class="anamnese-table">
    <tr>
        <th colspan="2">Fragen zur beruflichen Situation</th>
    </tr>
    <tr>
        <td>Name, Vorname</td>
        <td><?php echo $anamnese->client->surname . ' ' . $anamnese->client->name ?></td>
    </tr>
    <tr>
        <td>Geburtsdatum</td>
        <td><?php echo $anamnese->client->birthday ?></td>
    </tr>
    <tr>
        <td>Tel.</td>
        <td><?php echo $anamnese->client->phone ?></td>
    </tr>
    <tr>
        <td>Email</td>
        <td><?php echo $anamnese->client->e_mail ?></td>
    </tr>
    <tr>
        <td>Anschrift</td>
        <td><?php echo $form->textField($anamnese,'address') ?></td>
    </tr>
    <tr>
        <td>Was für einen Beruf führen Sie aktuell aus?</td>
        <td><?php echo $form->textField($anamnese,'profession') ?></td>
    </tr>
    <tr>
        <td>Welche Tätigkeiten beinhaltet ihre Arbeit hauptsächlich?</td>
        <td><?php echo $form->radioButtonList($anamnese,'activities', $anamnese->getActivitiesList()) ?></td>
    </tr>
    <tr>
        <th colspan="2">Fragen zu körperlicher Aktivität in der Freizeit</th>
    </tr>
    <tr>
        <td>Wie sieht Ihre körperliche Beanspruchung in der Freizeit oder auf dem Arbeitsweg aus?</td>
        <td><?php echo $form->radioButtonList($anamnese,'physical_demands', $anamnese->getPhysicalDemandsList()) ?></td>
    </tr>
    <tr>
        <th colspan="2">Fragen zur sportlichen Aktivität und Regenerationsmassnahmen</th>
    </tr>
    <tr>
        <td>Welche Sportarten betreiben Sie regelmässig?</td>
        <td><?php echo $form->textArea($anamnese,'sportarts') ?></td>
    </tr>
    <tr>
        <td>In welchem Umfang betreiben Sie die erwähnten Sportarten? (Anzahl Stunden pro Woche)</td>
        <td><?php echo $form->textField($anamnese,'sportarts_scope') ?></td>
    </tr>
    <tr>
        <td>In welcher Intensität betreiben sie die erwähnten Sportarten?</td>
        <td><?php echo $form->radioButtonList($anamnese,'sportarts_intencity', $anamnese->getSportartsIntencityList()) ?></td>
    </tr>
    <tr>
        <td>Wieviele Stunden schlafen sie normalerweise  pro Nacht?</td>
        <td>
			Unter der Woche:<br/>
			<?php echo $form->textField($anamnese,'sleep_week') ?><br/>
			Am Wochenende:<br/>
			<?php echo $form->textField($anamnese,'sleep_weekend') ?><br/>
		</td>
    </tr>
    <tr>
        <td>Wieviele Stunden Entspannung gönnen Sie sich pro Tag?</td>
        <td>
			Unter der Woche:<br/>
			<?php echo $form->textField($anamnese,'relaxation_week') ?><br/>
			Am Wochenende:<br/>
			<?php echo $form->textField($anamnese,'relaxation_weekend') ?><br/>
		</td>
    </tr>
    <tr>
        <td>Wieviele trainingsfreie Tage pro Woche haben Sie?</td>
        <td><?php echo $form->textField($anamnese,'training_dayoff') ?></td>
    </tr>
    <tr>
        <th colspan="2">Fragen zum Aktuellen Gesundheitsstand</th>
    </tr>
	<tr>
		<td>Sind sie zur Zeit in ärztlicher Behandlung?</td>
		<td>
			<?php echo $form->radioButtonList($anamnese,'medical_treatment', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?>
		</td>
	</tr>
	<tr>
		<td>Nehmen Sie regelmässig Medikmente?</td>
		<td>
			<?php echo $form->radioButtonList($anamnese,'taking_drugs', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?>
		</td>
	</tr>
    <tr>
        <td>
			Leiden Sie zurzeit an einer Verletzung?<br/>
			<br/><br/>
			<i>Falls ja:</i><br/>
			<ul class="labels">
				<li>Art der Verletzung?</li>
				<li>Verletzter Körperteil?</li>
				<li>Handelt es sich um eine chronische Verletzung?</li>
			</ul>
		</td>
        <td>
			<?php echo $form->radioButtonList($anamnese,'injury', $anamnese->getYesNoList()) ?><br/>
			<br/>
			<ul class="inputs">
				<li><?php echo $form->textField($anamnese,'injury_type') ?></li>
				<li><?php echo $form->textField($anamnese,'injury_bodypart') ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'injury_chronic', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
			</ul>
		</td>
    </tr>
    <tr>
        <td style="vertical-align: top;">
			Leiden Sie an einer der folgenden Krankheiten?<br/><br/>
			<ul class="labels">
				<li>Herzinfarkt (<0.5 Jahre)</li>
				<li>Arterielle Durchblutungsstörungen (Stadien III/IV)</li>
				<li>Raynauld Syndrom</li>
				<li>Vaskulitis</li>
				<li>Erkrankungen mit erhöhter Kälteempfinden</li>
				<li>Sensibilitätsstörungen</li>
				<li>Durchblutungsstörungen</li>
				<li>Nervenschädigung</li>
				<li>Replantation</li>
				<li>Schädigung peripherer Lymphgefässe</li>
				<li>Kältehämoglobinämie, -agglutin Krankheit, Kryogloulinämie</li>
				<li>Nieren- und Blasenaffektionen</li>
				<li>Schwere Herz- & Kreislauferkrankungen</li>
			</ul>
		</td>
        <td style="vertical-align: top;">
			<br/><br/>
			<ul class="inputs">
				<li><?php echo $form->radioButtonList($anamnese,'disease_heartattack', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_arterial_disorder', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_raynauld_syndrome', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_vasculitis', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_cold_sensitivity', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_sensory_disturbances', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_circulatory_disorder', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_nerve_damage', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_replantation', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_peripheral_lymphatics', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_hemoglobinemia', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_kidney_bladder', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
				<li><?php echo $form->radioButtonList($anamnese,'disease_heart_circulatory', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?></li>
			</ul>
		</td>
    </tr>
    <tr>
        <td>
			Leiden Sie an weiteren Problemen an Ihrem Bewegungsapparat?<br/>
			<br/>
			Falls ja, an welchen?
		</td>
        <td>
			<?php echo $form->radioButtonList($anamnese,'musculoskeletal_problems', $anamnese->getYesNoList(), array('separator' => '<span class="separator"></span>')) ?><br/>
			<br/>
			<?php echo $form->textField($anamnese,'musculoskeletal_problems_description') ?>
		</td>
    </tr>
    <tr>
        <th colspan="2">Trainingsziele</th>
    </tr>
    <tr>
        <td colspan="2">
			<?php echo $form->textArea($anamnese,'goals') ?>
		</td>
    </tr>
	<tr>
        <th colspan="2">Bemerkungen</th>
    </tr>
    <tr>
        <td colspan="2">
			<?php echo $form->textArea($anamnese,'comments') ?>
		</td>
    </tr>
</table>
<?php echo CHtml::submitButton('Save'); ?>
<?php $this->endWidget(); ?>

File: trainer-flutter/lib/models/anamnese.dart
class Anamnese {
  final int? clientId;
  String? address;
  String? profession;
  int? activities;         // 0=Sitzend, 1=Mässig, 2=Intensiv
  int? physicalDemands;    // 0-4 scale (PHP enum)
  String? sportarts;
  String? sportartsScope;
  int? sportartsIntensity; // 0=Leicht, 1=Moderat, 2=Anstrengend
  int? sleepWeek;          // hours per night
  int? sleepWeekend;       // hours per night
  String? relaxationWeek;
  String? relaxationWeekend;
  String? trainingDayoff;
  bool? injury;
  String? injuryType;
  String? injuryBodypart;
  bool? injuryChronic;
  bool? diseaseHeartattack;
  bool? diseaseArterialDisorder;
  bool? diseaseRaynaldSyndrome;
  bool? diseaseVasculitis;
  bool? diseaseColdSensitivity;
  bool? diseaseSensoryDisturbances;
  bool? diseaseCirculatoryDisorder;
  bool? diseaseNerveDamage;
  bool? diseaseReplantation;
  bool? diseasePeripheralLymphatics;
  bool? diseaseHemoglobinemia;
  bool? diseaseKidneyBladder;
  bool? diseaseHeartCirculatory;
  bool? musculoskeletalProblems;
  String? musculoskeletalProblemsDescription;
  String? comments;
  String? goals;
  bool? medicalTreatment;
  bool? takingDrugs;

  Anamnese({
    this.clientId,
    this.address,
    this.profession,
    this.activities,
    this.physicalDemands,
    this.sportarts,
    this.sportartsScope,
    this.sportartsIntensity,
    this.sleepWeek,
    this.sleepWeekend,
    this.relaxationWeek,
    this.relaxationWeekend,
    this.trainingDayoff,
    this.injury,
    this.injuryType,
    this.injuryBodypart,
    this.injuryChronic,
    this.diseaseHeartattack,
    this.diseaseArterialDisorder,
    this.diseaseRaynaldSyndrome,
    this.diseaseVasculitis,
    this.diseaseColdSensitivity,
    this.diseaseSensoryDisturbances,
    this.diseaseCirculatoryDisorder,
    this.diseaseNerveDamage,
    this.diseaseReplantation,
    this.diseasePeripheralLymphatics,
    this.diseaseHemoglobinemia,
    this.diseaseKidneyBladder,
    this.diseaseHeartCirculatory,
    this.musculoskeletalProblems,
    this.musculoskeletalProblemsDescription,
    this.comments,
    this.goals,
    this.medicalTreatment,
    this.takingDrugs,
  });

  static int? _parseInt(dynamic v) {
    if (v == null) return null;
    if (v is int) return v;
    if (v is String) return int.tryParse(v);
    if (v is double) return v.toInt();
    return null;
  }

  static bool? _parseBool(dynamic v) {
    if (v == null) return null;
    if (v is bool) return v;
    if (v is int) return v == 1;
    if (v is String) return v == '1' || v.toLowerCase() == 'true';
    return null;
  }

  factory Anamnese.fromJson(Map<String, dynamic> json) {
    return Anamnese(
      clientId: json['client_id'] is int
          ? json['client_id']
          : int.tryParse(json['client_id']?.toString() ?? ''),
      address: json['address']?.toString(),
      profession: json['profession']?.toString(),
      activities: _parseInt(json['activities']),
      physicalDemands: _parseInt(json['physical_demands']),
      sportarts: json['sportarts']?.toString(),
      sportartsScope: json['sportarts_scope']?.toString(),
      sportartsIntensity: _parseInt(json['sportarts_intencity']),
      sleepWeek: _parseInt(json['sleep_week']),
      sleepWeekend: _parseInt(json['sleep_weekend']),
      relaxationWeek: json['relaxation_week']?.toString(),
      relaxationWeekend: json['relaxation_weekend']?.toString(),
      trainingDayoff: json['training_dayoff']?.toString(),
      injury: _parseBool(json['injury']),
      injuryType: json['injury_type']?.toString(),
      injuryBodypart: json['injury_bodypart']?.toString(),
      injuryChronic: _parseBool(json['injury_chronic']),
      diseaseHeartattack: _parseBool(json['disease_heartattack']),
      diseaseArterialDisorder: _parseBool(json['disease_arterial_disorder']),
      diseaseRaynaldSyndrome: _parseBool(json['disease_raynauld_syndrome']),
      diseaseVasculitis: _parseBool(json['disease_vasculitis']),
      diseaseColdSensitivity: _parseBool(json['disease_cold_sensitivity']),
      diseaseSensoryDisturbances:
          _parseBool(json['disease_sensory_disturbances']),
      diseaseCirculatoryDisorder:
          _parseBool(json['disease_circulatory_disorder']),
      diseaseNerveDamage: _parseBool(json['disease_nerve_damage']),
      diseaseReplantation: _parseBool(json['disease_replantation']),
      diseasePeripheralLymphatics:
          _parseBool(json['disease_peripheral_lymphatics']),
      diseaseHemoglobinemia: _parseBool(json['disease_hemoglobinemia']),
      diseaseKidneyBladder: _parseBool(json['disease_kidney_bladder']),
      diseaseHeartCirculatory: _parseBool(json['disease_heart_circulatory']),
      musculoskeletalProblems: _parseBool(json['musculoskeletal_problems']),
      musculoskeletalProblemsDescription:
          json['musculoskeletal_problems_description']?.toString(),
      comments: json['comments']?.toString(),
      goals: json['goals']?.toString(),
      medicalTreatment: _parseBool(json['medical_treatment']),
      takingDrugs: _parseBool(json['taking_drugs']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (clientId != null) 'client_id': clientId,
      if (address != null) 'address': address,
      if (profession != null) 'profession': profession,
      'activities': activities ?? 0,
      'physical_demands': physicalDemands ?? 0,
      if (sportarts != null) 'sportarts': sportarts,
      if (sportartsScope != null) 'sportarts_scope': sportartsScope,
      'sportarts_intencity': sportartsIntensity ?? 0,
      'sleep_week': sleepWeek ?? 0,
      'sleep_weekend': sleepWeekend ?? 0,
      if (relaxationWeek != null) 'relaxation_week': relaxationWeek,
      if (relaxationWeekend != null) 'relaxation_weekend': relaxationWeekend,
      if (trainingDayoff != null) 'training_dayoff': trainingDayoff,
      if (injury != null) 'injury': injury! ? 1 : 0,
      if (injuryType != null) 'injury_type': injuryType,
      if (injuryBodypart != null) 'injury_bodypart': injuryBodypart,
      if (injuryChronic != null) 'injury_chronic': injuryChronic! ? 1 : 0,
      if (diseaseHeartattack != null)
        'disease_heartattack': diseaseHeartattack! ? 1 : 0,
      if (diseaseArterialDisorder != null)
        'disease_arterial_disorder': diseaseArterialDisorder! ? 1 : 0,
      if (diseaseRaynaldSyndrome != null)
        'disease_raynauld_syndrome': diseaseRaynaldSyndrome! ? 1 : 0,
      if (diseaseVasculitis != null)
        'disease_vasculitis': diseaseVasculitis! ? 1 : 0,
      if (diseaseColdSensitivity != null)
        'disease_cold_sensitivity': diseaseColdSensitivity! ? 1 : 0,
      if (diseaseSensoryDisturbances != null)
        'disease_sensory_disturbances': diseaseSensoryDisturbances! ? 1 : 0,
      if (diseaseCirculatoryDisorder != null)
        'disease_circulatory_disorder': diseaseCirculatoryDisorder! ? 1 : 0,
      if (diseaseNerveDamage != null)
        'disease_nerve_damage': diseaseNerveDamage! ? 1 : 0,
      if (diseaseReplantation != null)
        'disease_replantation': diseaseReplantation! ? 1 : 0,
      if (diseasePeripheralLymphatics != null)
        'disease_peripheral_lymphatics': diseasePeripheralLymphatics! ? 1 : 0,
      if (diseaseHemoglobinemia != null)
        'disease_hemoglobinemia': diseaseHemoglobinemia! ? 1 : 0,
      if (diseaseKidneyBladder != null)
        'disease_kidney_bladder': diseaseKidneyBladder! ? 1 : 0,
      if (diseaseHeartCirculatory != null)
        'disease_heart_circulatory': diseaseHeartCirculatory! ? 1 : 0,
      if (musculoskeletalProblems != null)
        'musculoskeletal_problems': musculoskeletalProblems! ? 1 : 0,
      if (musculoskeletalProblemsDescription != null)
        'musculoskeletal_problems_description':
            musculoskeletalProblemsDescription,
      if (comments != null) 'comments': comments,
      if (goals != null) 'goals': goals,
      if (medicalTreatment != null)
        'medical_treatment': medicalTreatment! ? 1 : 0,
      if (takingDrugs != null) 'taking_drugs': takingDrugs! ? 1 : 0,
    };
  }
}
